### DISCUSSION TITLE
디비에서 전체 데이터를 통해서 가져오기 VS 디비 자체에서 가공해서 가져오기

### DISCUSSION THREAD

**Discussion by Aleexender** [SCBJ-7/SCBJ-BE#61] [Ideas]
제가 지금 쿼리문(Query 어노테이션)을 작성하다가 고민이 생겨서 이렇게 작성을 하게되었어요!

제가 지금 착각을해서 API를 하나밖에 안만들어놨는데 ㅠㅠ 

이용 예정 쪽 API 를 수정할려고 합니다! 근데  여기서 궁금한게 DB 상에서 현재 날짜를 통해서 현재날짜 이전인 데이터들을 가공해서 가져오는게 나을까요? 아니면 전부다 가져와서 비지니스 로직에서 가공해서 보내주는게 나을까요???? 

깔끔해보일거같은건 전자이긴한데... 성능상 이슈가 좀 생길수도 있을거같다는 생각이 문뜩드네용!

다른분들의 의견이 궁금합니다!

**Comment by Kwonminwoo** [comment]
디비와 통신하는 부하가 적을 수록 이득이라고 생각합니다. 데이터가 커질 수록 모두 가져오는 건 부하가 심해지니 피하는게 좋을 것 같아요!!

**Reply by Aleexender** [reply]
그럼 후자를 추천하시는건가용???

**Reply by Aleexender** [reply]
아 전자군요! 알겠습니다!

**Comment by YurimYang** [comment]
저도 민우님이랑 동일하게 생각해요 !! 최대한 가공해서 가져오는게 좋을 것같습니다 ~
